lib/loaders/createConnectionLoaderClass.ts

```diff
@@ -187,9 +189,22 @@ export const createConnectionLoaderClass = <TResult>(config: {
             }
           });
 
+          let extendedParser;
+
+          if (query.parser) {
+            const parser = query.parser as unknown as AnyZodObject;
+
+            extendedParser = parser.extend({
+              key: z.union([z.string(), z.number()]),
```

Comment:
Wasn't 100% sure what are the possible values of a key, so made an assumption that either a number or a string.